railties/test/application/initializers/i18n_test.rb

```diff
@@ -45,6 +45,13 @@ def assert_no_fallbacks
       assert_equal :de, I18n.default_locale
     end
 
+    test "generating application template with locale" do
+      load_app
+      
+      application_layout = File.read("#{app_path}/app/views/layouts/application.html.erb")
+      assert_includes application_layout, "<html lang=\"<%= I18n.locale %>\">"
```

Comment:
Isn't this testing the same as the test in app_generator_test? I think what you want is a test that renders the page, so it would be closer to this test here https://github.com/rails/rails/blob/09d55b302266cf002a4b307f8d37a105d2838a18/railties/test/application/content_security_policy_test.rb#L19-L38

Reply:
Hmm sorry, my test was looking at the actual `application.html.erb` file, not at the HTML output. Thanks for sharing that example, I'll write a proper test for this.

Reply:
@rafaelfranca before I update this, is `railties/test/application/initializers/i18n_test.rb` still the best place for this test?

Reply:
Yup, you can still add the test there 👍🏼

Reply:
Added two new tests. Please have a look and let me know if I'm on the right track